sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutorUtil.java

```diff
@@ -45,6 +46,44 @@ static ServerOperationBatchRequest createBatchRequest(List<CosmosItemOperation>
             BatchRequestResponseConstants.MAX_OPERATIONS_IN_DIRECT_MODE_BATCH_REQUEST);
     }
 
+    static ServerOperationBatchRequest createBatchRequest(List<CosmosItemOperation> operations, String partitionKeyRangeId, int maxMicroBatchPayloadSizeInBytes, PartitionSplitNotifier partitionSplitNotifier) {
+
+        return PartitionKeyRangeServerBatchRequest.createBatchRequest(
+            partitionKeyRangeId,
+            operations,
+            maxMicroBatchPayloadSizeInBytes,
+            BatchRequestResponseConstants.MAX_OPERATIONS_IN_DIRECT_MODE_BATCH_REQUEST, partitionSplitNotifier);
+    }
+
+    static boolean isTransientFailure(int statusCode, int substatusCode) {
+        return statusCode == HttpConstants.StatusCodes.GONE
+            || statusCode == HttpConstants.StatusCodes.SERVICE_UNAVAILABLE
+            || statusCode == HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR
+            || statusCode == HttpConstants.StatusCodes.REQUEST_TIMEOUT
+            || (statusCode == HttpConstants.StatusCodes.NOTFOUND && substatusCode == HttpConstants.SubStatusCodes.READ_SESSION_NOT_AVAILABLE)
+            || (statusCode == HttpConstants.StatusCodes.TOO_MANY_REQUESTS);
+
+    }
+
+    static boolean isTransientFailure(Exception e) {
+        if (e instanceof CosmosException) {
```

Comment:
Nit: You could get a `CosmosException` instance as this `Utils.as(e, CosmosException.class)`

Reply:
changed